### PR TITLE
ddl:enhance ddl unit tests in tidb

### PR DESCRIPTION
**Pull request by ciscoxll**
Some tests have been added.
* ErrWrongFkDef                                            = 1239
* ErrTooManyKeyParts                                    = 1070
* ErrMultiplePriKey                                           = 1068
* ErrWrongColumnName                                = 1166
* ErrInvalidOnUpdate                                       = 1294 
* ErrCantRemoveAllFields                              = 1090
* ErrBlobKeyWithoutLength                          = 1170
* ErrTooLongKey                                                  = 1071
* ErrWrongNameForIndex                                  = 1280

### CODE REVIEW

**Comment by shenli** [comment]
@ciscoxll Please fix the CI.

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -323,7 +323,7 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef) (
 				removeOnUpdateNowFlag(col)
 			case ast.ColumnOptionOnUpdate:
 				// TODO: Support other time functions.
-				if !expression.IsCurrentTimestampExpr(v.Expr) {
+				if expression.IsCurrentTimestampExpr(v.Expr) {
```
Comment (zimulala) [inline ddl/ddl_api.go]: Why change this check?

Comment (ciscoxll) [reply ddl/ddl_api.go]: This code was not tested before, logic is the opposite.

**Comment by shenli** [comment]
LGTM

**Pull request merged**